# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/test/browser/browser_test_local_network_trackers.js
# Commit: 1d0091c56339
# Full Hash: 1d0091c5633982e358737a96ed8af1a2be00bf61
# Author: smayya <smayya@mozilla.com>
# Date: 2025-07-27 09:20:01
# Regressor Bug: 1960627
# File Overlap Count: 1
# Description:
#   Bug 1960627 - add LNAPermissionRequest class for dispatching permission prompt requests. r=necko-reviewers,emz,kershaw
#   
#   LNAPermissionRequest is used for dispatching localhost and local-network permission prompts to the user.
#   Additionally, this patch also implements the transaction retry mechanism based on prompt action by the user.
#   This patch also modifies existing unit tests to validate the transaction retry mechanism.
# ==============================================================================

diff -r 0981f147243d -r 1d0091c56339 netwerk/test/browser/browser_test_local_network_trackers.js
--- a/netwerk/test/browser/browser_test_local_network_trackers.js	Sat Jul 26 21:51:08 2025 +0000
+++ b/netwerk/test/browser/browser_test_local_network_trackers.js	Sat Jul 26 21:51:08 2025 +0000
@@ -51,6 +51,12 @@
       ["network.websocket.max-connections", 1000],
       ["network.lna.block_trackers", true],
       ["network.lna.address_space.public.override", "127.0.0.1:4443"],
+      ["network.lna.blocking", true],
+      // always select allow actions for user prompts
+      ["network.localhost.prompt.testing", true],
+      ["network.localnetwork.prompt.testing", true],
+      ["network.localhost.prompt.testing.allow", true],
+      ["network.localnetwork.prompt.testing.allow", true],
     ],
   });
 